### PR TITLE
Add endpoint for /internal/amiusage

### PR DESCRIPTION
**Pull request by rafvasq**
Implemented an endpoint “GET /internal/ami-usage” which lists out all of the AMIs and their usage and included unit test

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -577,6 +577,22 @@ api.declare({
...
+api.declare({
+  method: 'get',
+  route: '/internal/ami-usage',
+  name: 'amiUsage',
+  title: 'See the list of AMIs and their usage',
+  stability: API.stability.experimental,
+  scopes: [['ec2-manager:internals']],
+  description: 'Lists out all of the AMIs and their usage',
+}, async function(req, res) {
+  let amiUsage = await this.state.listAmiUsage();
```
Comment (walac) [inline lib/api.js]: It has been a standard in this project, so it is okay, but bear in mind that in general, variables should be `const` by default and we only use `let` for cases where we have good reasons to modify them (like the control variable in a `for` loop).

File: lib/api.js [hunk 2]
```diff
@@ -577,6 +577,22 @@ api.declare({
 });
 
 /**
+ * List AMIs and their usage
+ */
+api.declare({
+  method: 'get',
+  route: '/internal/ami-usage',
+  name: 'amiUsage',
+  title: 'See the list of AMIs and their usage',
```
Comment (walac) [inline lib/api.js]: nit: s/See/Show/

File: lib/api.js [hunk 3]
```diff
@@ -577,6 +577,22 @@ api.declare({
...
+  stability: API.stability.experimental,
+  scopes: [['ec2-manager:internals']],
```
Comment (walac) [inline lib/api.js]: I am not very much familiar with the semantics of the service. Therefore I will leave review of scopes management to @jhford

File: lib/api.js (hunk 1 above)
Comment (rafvasq) [reply lib/api.js]: Good to know, thanks for the info!

File: lib/api.js (hunk 2 above)
Comment (rafvasq) [reply lib/api.js]: I would go with "show" as well but other GET methods use "See the lists ..." so I decided to be consistent

File: lib/api.js (hunk 1 above)
Comment (walac) [reply lib/api.js]: Good point.

File: lib/api.js (hunk 2 above)
Comment (jhford) [reply lib/api.js]: Let's stick with consistency.

File: lib/api.js (hunk 3 above)
Comment (jhford) [reply lib/api.js]: This is an endpoint that's only useful to the taskcluster team, and as a /internals/ endpoint, i'd like to keep it locked down like this.  This is the right scope.

**Pull request merged**